### PR TITLE
Adding the ability to pass in IV

### PR DESCRIPTION
**Pull request by bhariharan**
Added new methods that take an additional parameter, `iv`. Existing behavior and APIs remain the same.

### CODE REVIEW

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">📱 Needs testing on a real device if change is non-trivial.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_PR-Check">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java [hunk 1]
```diff
@@ -142,6 +154,18 @@ public static String decrypt(String data, String key) {
      * @return Decrypted data.
      */
     public static String decrypt(byte[] data, String key) {
+        return decrypt(data, key, new byte[16]);
```
Comment (Songroid) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: Why do we use `new byte[16]` here as a parameter of the constructor of `IvParameterSpec`?

Comment (bhariharan) [reply libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java]: @Songroid It was [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1611/files#diff-d64d9d238bf5bb327ddd3e159882e88bL338) before. Now it's passed in from the top. These methods exist for backward compatibility.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/aed0f086410a35be6f87eef71c7d9b63aad937c6/libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java#L248">libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java#L248</a> - Consider using <code>apply()</code> instead; <code>commit</code> writes its data to persistent storage immediately, whereas <code>apply</code> will handle it in the background</td>
    </tr>
  </tbody>
</table>

# Tests results for SalesforceSDK

<p align="right" data-meta="generated_by_SalesforceSDK">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by Songroid** [comment]
Nope looks good. Wait for Sasha's test on the new encoded string.

**Pull request merged**